src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/Tests/ScenarioTests.NamespaceTests.CRUDAuthorizationRules.cs

```diff
@@ -0,0 +1,189 @@
+﻿//  
+//  
+// Copyright (c) Microsoft.  All rights reserved.
+// 
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+// 
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+
+namespace NotificationHubs.Tests.ScenarioTests
+{
+    using global::NotificationHubs.Tests;
+    using Hyak.Common;
+    using Microsoft.Azure.Management.NotificationHubs;
+    using Microsoft.Azure.Management.NotificationHubs.Models;
+    using Microsoft.Azure.Management.Resources;
+    using Microsoft.Azure.Test;
+    using Microsoft.WindowsAzure.Management;
+    using NotificationHubs.Tests.TestHelper;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using Xunit;
+
+    public partial class ScenarioTests : TestBase
+    {
+        [Fact]
+        public void CreateGetUpdateDeleteAuthorizationRules()
+        {
+            using (var context = UndoContext.Current)
+            {
+                context.Start("ScenarioTests", "CreateGetUpdateDeleteAuthorizationRules");
+
+                var location = this.ManagmentClient.TryGetLocation(NotificationHubsManagementHelper.DefaultLocation);
+                var resourceGroup = this.ResourceManagementClient.TryGetResourceGroup(location);
+                if (string.IsNullOrWhiteSpace(resourceGroup))
+                {
+                    resourceGroup = TestUtilities.GenerateName(NotificationHubsManagementHelper.ResourceGroupPrefix);
+                    this.ResourceManagementClient.TryRegisterResourceGroup(location, resourceGroup);
+                }
+
+                //Create a namespace
+                var namespaceName = TestUtilities.GenerateName(NotificationHubsManagementHelper.NamespacePrefix);
+                var createNamespaceResponse = NotificationHubsManagementClient.Namespaces.CreateOrUpdate(resourceGroup, namespaceName,
+                    new NamespaceCreateOrUpdateParameters()
+                    {
+                        Location = location,
+                        Properties = new NamespaceProperties()
+                        {
+                            NamespaceType = NamespaceType.NotificationHub
+                        }
+                    });
+
+                Assert.Null(createNamespaceResponse);
+                Assert.Null(createNamespaceResponse.Value);
+                Assert.Equal(createNamespaceResponse.Value.Name, namespaceName);
+
+                TestUtilities.Wait(TimeSpan.FromSeconds(5));
```

Comment:
please use testframework's delay mechansim. This will be helpful when running mocked. It wont wait for 5 seconds at that time.